_pytest/test_multitenancy.py

```diff
@@ -28,110 +34,136 @@ def app(component_builder):
         'port': -1,  # unused in test app
         "backend": "mitie",
         "path": os.path.join(root_dir, "test_models"),
-        "data": os.path.join(root_dir, "data/demo-restaurants.json"),
-        "server_model_dirs": {
-            "one": "test_model_mitie",
-            "two": "test_model_mitie_sklearn",
-            "three": "test_model_spacy_sklearn",
-        }
+        "data": os.path.join(root_dir, "data/demo-restaurants.json")
     }
-    config = RasaNLUConfig(cmdline_args=_config)
-    application = create_app(config, component_builder)
-    return application
+    url = '127.0.0.1'
+    port = 5000
+    train_models(component_builder)
+    pid = os.fork()
+    if pid == 0:
+        sem.acquire()
+        config = RasaNLUConfig(cmdline_args=_config)
+        rasa = RasaNLU(config, component_builder)
+        sem.release()
+        rasa.app.run(url, port)
+        rasa.data_router.__del__()
```

Comment:
I'm sorry but I didn't get it: overriding `__del__()` is the way to go for cleaning objects, isn't it? So why rename the function? `del` never explicitly call `__del__()` but the garbage collector that cleans objects marked for deletion is supposed to. I just call it explicitly because there is no guaranty that the garbage collector will make its pass before killing the process.